repos/system_upgrade/el7toel8/libraries/rhsm.py

```diff
@@ -93,81 +95,122 @@ def _no_op(*args, **kwargs):
-            result = context.call(['subscription-manager', 'list', '--consumed'], split=False)
-            rhsm_info.attached_skus = _RE_SKU_CONSUMED.findall(result['stdout'])
+    with _handle_rhsm_exceptions():
+        result = context.call(['subscription-manager', 'list', '--consumed'], split=False)
+        return _RE_SKU_CONSUMED.findall(result['stdout'])
 
 
-@with_rhsm
-def get_available_repo_uids(context, rhsm_info):
+def get_available_repo_ids(context):
     """
-    Retrieves all available repositories names from the subscription manager.
+    Retrieve repo ids of all the repositories available through the subscription-manager.
 
     :param context: An instance of a mounting.IsolatedActions class
     :type context: mounting.IsolatedActions class
-    :param rhsm_info: An instance of a RHSMInfo derived model.
-    :type rhsm_info: RHSMInfo derived model
+    :return: Repositories that are available to the current system through the subscription-manager
+    :rtype: List(string)
     """
-    if not rhsm_info.available_repos:
-        with _handle_rhsm_exceptions():
-            result = context.call(['subscription-manager', 'repos'], split=False)
-            rhsm_info.available_repos = _RE_REPO_UID.findall(result['stdout'])
+    result = context.call(['yum', 'repoinfo'])
+    _inhibit_on_duplicate_repos(result['stderr'])
+    all_repos = list(_get_repos(result['stdout']))
+    available_repos = [repo.id for repo in all_repos if repo.file == _DEFAULT_RHSM_REPOFILE]
+    api.current_logger().info('The following repoids are available through RHSM:\n{0}'
+                              .format('\n'.join(available_repos)))
+    return available_repos
 
 
-@with_rhsm
-def get_enabled_repo_uids(context, rhsm_info):
+def _inhibit_on_duplicate_repos(repos_raw_stderr):
     """
-    Retrieves all enabled repositories names from the subscription manager.
+    Inhibit the updgrade if a single repo is defined multiple times.
 
-    :param context: An instance of a mounting.IsolatedActions class
-    :type context: mounting.IsolatedActions class
-    :param rhsm_info: An instance of a RHSMInfo derived model.
-    :type rhsm_info: RHSMInfo derived model
+    When that happens, it not only shows miscofigured system, but then we can't get the details of all the available
+    repos as well.
     """
-    if not rhsm_info.enabled_repos:
-        with _handle_rhsm_exceptions():
-            result = context.call(['subscription-manager', 'repos', '--list-enabled'], split=False)
-            rhsm_info.enabled_repos = _RE_REPO_UID.findall(result['stdout'])
+    duplicates = []
+    for duplicate in re.findall(
+            r'Repository ([^\s]+) is listed more than once', repos_raw_stderr, re.DOTALL | re.MULTILINE):
+        duplicates.append(duplicate)
+
+    if not duplicates:
+        return
+
+    api.current_logger().warn('The following repoids are defined multiple times:\n{0}'
+                              .format('\n'.join(duplicates)))
```

Comment:
I prefer to use the separator we are using on several other places to be consistent and improve readability, e.g. here: https://github.com/oamg/leapp-repository/blob/master/repos/system_upgrade/el7toel8/actors/vsftpdconfigcheck/libraries/library.py